docs/docs/Tutorials/chat-with-files.md

```diff
@@ -0,0 +1,173 @@
+
+## Prerequisites
+
+- [A running Langflow instance](/get-started-installation)
+- [An OpenAI API key](https://platform.openai.com/api-keys)
+
+## Create a file loader chat flow
+
+1. In Langflow, click **New Flow**, and then select the **Basic prompting** template.
+2. Add your **OpenAI** API key to the **Language Model** component.
+3. To confirm that the chatbot responds as expected, open the <Icon name="Play" aria-hidden="true" /> **Playground** and then ask a question.
+The LLM should respond as the **Prompt** component specifies.
+With the basic flow responding correctly, modify the **Prompt** component to accept additional inputs.
+4. In the **Prompt** component, open the **Template** field, and replace the current prompt with the following text:
+    ```text
+    ChatInput:
+    {chat-input}
+    File:
+    {file}
+    ```
+    The **Prompt** component adds ports to accept new inputs for each value with curly braces, in this case, a chat input and a file.
+
+    :::tip
+    The port names are for illustrative purposes. You don't have to name them after the component they're connected to.
+    :::
+
+5. Add a [File component](/components-data#file) to the flow, and connect the **Raw Content** output to the Prompt component's `file` input.
+
+    Your flow should look like this:
+
+    ![File loader chat flow](/img/tutorial-chat-file-loader.png)
+
+    The flow is complete.
+    If you'd like, add files to the File component and chat with it within the Langflow IDE.
+    In the next section, you will load files and chat with your flow from a Python application.
+
+## Send requests to your flow from a Python application
+
+With your flow running locally, send a request to `POST /run` to load a file to your flow, send a chat message, and get a result back.
+
+:::tip
+To easily construct file upload requests in Python, JavaScript, and curl templates, check out the [Langflow File Upload Utility](https://langflow-file-upload-examples.onrender.com).
+:::
+
+1. To construct the request, gather the following values from Langflow.
+
+    * `LANGFLOW_SERVER_ADDRESS`: The default value is `127.0.0.1:7860`.
+    * `FLOW_ID`: The UUID of your flow, or the endpoint name you've chosen.
+    * File Component name: To find the ID of your file component, in the File component, click **Controls**. For this example, the component name is `File-KZP68`.
+    * Input value: This is the message you want to send to the Chat Input of your flow, such as `Evaluate this resume for a job opening in my Marketing department.`
+    * File path: The path to the local file you want to load with your request. This example is loading `fake-resume.txt` from the same file location as the script.
+    * Session ID: Optional. For more information, see [session ID](/session-id).
+    * Langflow API key: Required. To create an API key, see [API keys](/configuration-api-keys).
+
+2. Replace the values in the script below.
+
+    <details open>
+        <summary>Python</summary>
```

Comment:
```suggestion
```

Having details set to open by default defeats the purpose of details. Unless you plan to add Java and HTTP examples, I would just leave out the details wrapper.